src/github_proxy.py

```diff
@@ -46,14 +54,29 @@ def publish_pr_comment(self, build):
             logs_url=build.get_logs_url(),
         )
 
-        # initialize client before logging to ensure GitHub attributes are populated
-        gh_client = self._get_client()
+        repo = self._get_repo()
         LOG.debug('Publishing PR Comment: repo=%s/%s, pr_id=%s, comment=%s',
                   self._github_owner, self._github_repo, build.get_pr_id(), pr_comment)
-
-        repo = gh_client.get_user(self._github_owner).get_repo(self._github_repo)
         repo.get_pull(build.get_pr_id()).create_issue_comment(pr_comment)
 
+    def delete_previous_comments(self, build):
```

Comment:
I don't see any unit tests for this new method. Can you add them? I know we're still above the 85% coverage threshold, but I had these files at 100% coverage so want to maintain a high quality bar.

Reply:
Done, we're back to 100%.

Note: I'm not really familiar with python testing, doing almost only javascript/typescript for the last few years... tried my best here...